client/services/cardKeyService.js

```diff
@@ -115,7 +115,20 @@ module.exports = [
           while (!finished) {
             const { value } = await $rootScope.serialReader.read();
             for (let i = 0; i < value.length; i++) {
-              if (value[i] == 0xbb) {
+              // First byte in a message should be 170, otherwise ignore and keep on going
+              if (message.length === 0 && value[i] !== 170) {
+                continue;
+              }
+              // Second byte in a message should be 255, otherwise discard and keep on going
+              if (message.length === 1 && value[i] !== 255) {
+                // If value is 170, treat it as the first value, and keep on. Otherwise discard
+                if (value[i] !== 170) {
+                  message.length = 0;
+                }
+                continue;
+              }
+
+              if (message.length > 3 && message.length >= message[2] + 4) {
```

Comment:
Is this some known fact of the reader/cards or is this just a result from testing different cards?

Reply:
Just quick brute force to make it work properly with ids containing 0xbb. Not sure where the initial code here originates, and since the card readers are just random 10 year old china stuff I don't know of any datasheets.

This now reads ids in the same way as web-nfc, and in case stuff doesn't work properly, it will be detected by the checksum calculation.

Reply:
We don't need this patch, but of we should use this together with web-nfc, they has to read the same cards, and this makes sure that is the case. (Web-nfc doesn't work for payment cards tho., so we should probabaly ignore them)